crates/primitives/src/bytecode.rs

```diff
@@ -75,11 +82,15 @@ impl Bytecode {
         }
     }
 
-    /// Create new raw Bytecode with hash
+    /// Creates a new raw Bytecode with the given hash.
     ///
     /// # Safety
-    /// Hash need to be appropriate keccak256 over bytecode.
+    ///
+    /// The given `hash` has to be equal to the [`keccak256`] hash of the given `bytecode`.
+    #[inline]
+    #[cfg_attr(debug_assertions, track_caller)]
```

Comment:
Why is this added, in essence, `unsafe` is here for a reason user of this should be careful what he does

Reply:
Right, it's just an extra check in debug mode. I can remove it if you want

Reply:
Let us remove it, I would assume there is a possibility for one to intentionally switch the bytecode of a different hash.